docs/src/main/sphinx/develop/table-functions.rst

```diff
@@ -0,0 +1,133 @@
+===============
+Table functions
+===============
+
+Table function declaration
+--------------------------
+
+Trino supports adding custom table functions. They are declared by connectors
```

Comment:
So ... thats kinda weird .. so users have to maintain a fork of the connector? It cant be a separate plugin that is somehow associated with a specific connector?

Reply:
No, it doesn't require a fork or a separate plugin. Any existing connector can expose table functions. Soon we'll have the query pass-through feature, which is a good examples of (existing) connectors registering polymorphic table functions.

Reply:
Generally, I believe that we will be adding a lot of custom table functions to Trino. Of course, we will have to judge if the function is generally useful. If someone need a function that we find "controversial", they might have to fork.

Reply:
okay ... maybe we should say something like.. want another table function added to trino .. file a ticket or PR and contact us ... although thats kinda odd for the docs.. not sure at this stage how we should proceed .. if there are not currently usable table functions maybe we should merge..

Reply:
Yeah, we don't write things like "want another connector...file a ticket..." in the docs. 

At the moment, we have a bunch of table functions in review. They should be in soon. When we merge them, we'll document them, and they will serve as a good example of what people have to do if they want a new table function in Trino.

Also, we will likely have some publications around table functions.